scripts/update-charts-dep.sh

```diff
@@ -41,7 +41,7 @@ echo "Updating all Charts..."
 for chart in "${charts[@]}"
 do
     echo "---=== Updating $chart ===---"
-    helm dep up "$chart" --skip-refresh
+    helm dep up "$chart"
```

Comment:
Please ensure that you run `helm repo update` prior to running this.

Reply:
I'm not sure why we are running `helm dep update` for every dependency, instead of high-level charts, i.e. `bof`, mojaloop, etc. Isn't it the job of Helm to do manage dependencies?

For me it looks like the same if we were parsing `package.json` with a script and trying to install all dependencies manually instead of calling `npm install` for the whole package.

Reply:
Hi @kirgene, that's because we want all the charts to get published instead of only high-level charts.
So we need the dependencies to be populated in the respective charts folder for all the sub charts also.

Reply:
Hi @vijayg10 , am not sure I follow. All (sub)dependencies of high-level charts are published, otherwise, what's the point of the "dependency" feature in charts.

Reply:
Am not aware of any dependency problems in helm.
Could you describe how you performed those tests.

Reply:
@mdebarros @vijayg10 please see the following repo for an example of using helm dependencies:
https://github.com/pm4ml/mojaloop-payment-manager-helm/blob/master/mojaloop-payment-manager/Chart.yaml

Reply:
I can see that most [bof chart](https://github.com/mojaloop/charts/blob/2b6cda0a04e279c389932e65ccf5b9d9111c3bd0/mojaloop/bof/Chart.yaml#L25) dependencies are still 1.0.0, despite being updated frequently.
This obviously prevents fetching up-to-date changes from charts (and dependencies).

The correct way of handling these updates is:

1. Modify chart, for example, bump docker image version in [values file](https://github.com/mojaloop/charts/blob/master/mojaloop/reporting-hub-bop-settlements-ui/values.yaml)
2. Bump [Chart version](https://github.com/mojaloop/charts/blob/2b6cda0a04e279c389932e65ccf5b9d9111c3bd0/mojaloop/reporting-hub-bop-settlements-ui/Chart.yaml#L6)
3. Bump dependency version in [the dependent Chart](https://github.com/mojaloop/charts/blob/2b6cda0a04e279c389932e65ccf5b9d9111c3bd0/mojaloop/bof/Chart.yaml#L94)
4. Bump [dependent Chart](https://github.com/mojaloop/charts/blob/2b6cda0a04e279c389932e65ccf5b9d9111c3bd0/mojaloop/bof/Chart.yaml#L18) version

Reply:
This process is correct:

> Modify chart, for example, bump docker image version in values file
> Bump Chart version
> Bump dependency version in the dependent Chart
> Bump dependent Chart version

However, if we are publishing, we need to build the dependency tree manually, as this is not currently handled by Helm itself, ref: https://github.com/helm/helm/issues/2247.